### DISCUSSION TITLE
Introduce consistent behaviour for focus and keyboard input

### DISCUSSION THREAD

**Discussion by khirsig** [danvick/flutter_form_builder#1297] [Ideas]
Based on my own experience I've created together with @grundid a demo app, to showcase the differences between a modified FormBuilder version that implements consistent behaviour for focus and keyboard input and the regular FormBuilder behaviour.

In this demo app we have created two tabs with identical FormBuilder widget configurations. However on the left tab we used FormBuilder widgets that include our focus and keyboard input modifications. You can switch any time between those two forms and compare the focus and keyboard input behaviour. Below the form you can request focus for any of the form elements via a button.
Please have a look at the source code at my prebuild demo-app: https://khirsig.github.io/form_builder_demo/
Source Code for the demo: https://github.com/khirsig/form_builder_demo

The intention of this demo is to be able to focus every input component, and this means that the InputDecorator recieves the focus for the component. At the same time you can switch between mouse/touch input and keyboard input at any time. 

What do you think about this idea and would you accept pull requests concerning this topic?

**Comment by Adam-Langley** [comment]
I have just recently come across this exact issue.
On Windows desktop, the user is tab-traversing fields to get to a specific input, if they encounter a date picker, it pops up, blocking them from proceeding any further. Does not appear to follow expected UX conventions.

**Comment by richardhaven** [comment]
great demo

any interest in making a PR for these? I would have to stop using date-time to avoid it

Cheers

**Comment by deandreamatias** [comment]
> Based on my own experience I've created together with @grundid a demo app, to showcase the differences between a modified FormBuilder version that implements consistent behaviour for focus and keyboard input and the regular FormBuilder behaviour.
> 
> In this demo app we have created two tabs with identical FormBuilder widget configurations. However on the left tab we used FormBuilder widgets that include our focus and keyboard input modifications. You can switch any time between those two forms and compare the focus and keyboard input behaviour. Below the form you can request focus for any of the form elements via a button. Please have a look at the source code at my prebuild demo-app: https://khirsig.github.io/form_builder_demo/ Source Code for the demo: https://github.com/khirsig/form_builder_demo
> 
> The intention of this demo is to be able to focus every input component, and this means that the InputDecorator recieves the focus for the component. At the same time you can switch between mouse/touch input and keyboard input at any time.
> 
> What do you think about this idea and would you accept pull requests concerning this topic?

Sorry for delay. I am ashamed to have left this thread for a year without a response.
I totally open to anyone open PR. I open to discusse and find the best way to increase the DX and UX for this package. 
Maybe will be a slow process, but the important is accomplish the best and consensual result with community.

Currently I also review a [refactor on validator package](https://github.com/flutter-form-builder-ecosystem/form_builder_validators/pull/129)

**Comment by deandreamatias** [comment]
I'm start a PR with this solution base
Can take a look here #1453

**Comment by deandreamatias** [comment]
Related with #1292, #845 and #1287

**Comment by deandreamatias** [comment]
Generated prerelease version [10.0.0-dev.1](https://pub.dev/packages/flutter_form_builder/versions/10.0.0-dev.1) that solve some issues
